### ISSUE TITLE
fresh install (web / shell) fails with error

### ISSUE THREAD

**Issue by arosenhagen** [mageplus/mageplus#114]
Doing a completely fresh install fails with the following error report. Seems as if the install script calls the upgrade routines?!

``` php
a:5:{i:0;s:218:"Error in file: "/XXX/app/code/core/Mage/Sales/sql/sales_setup/install-1.6.0.0.php" - SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'sales_flat_order' already exists";i:1;s:1088:"#0 /XXX/app/code/core/Mage/Core/Model/Resource/Setup.php(645): Mage::exception('Mage_Core', 'Error in file: ...')
#1 /XXX/app/code/core/Mage/Core/Model/Resource/Setup.php(421): Mage_Core_Model_Resource_Setup->_modifyResourceDb('install', '', '1.6.1.0')
#2 /XXX/app/code/core/Mage/Core/Model/Resource/Setup.php(327): Mage_Core_Model_Resource_Setup->_installResourceDb('1.6.1.0')
#3 /XXX/app/code/core/Mage/Core/Model/Resource/Setup.php(235): Mage_Core_Model_Resource_Setup->applyUpdates()
#4 /XXX/app/code/core/Mage/Core/Model/App.php(417): Mage_Core_Model_Resource_Setup::applyAllUpdates()
#5 /XXX/app/code/core/Mage/Core/Model/App.php(343): Mage_Core_Model_App->_initModules()
#6 /XXX/app/Mage.php(684): Mage_Core_Model_App->run(Array)
#7 /XXX/index.php(87): Mage::run('', 'store')
#8 {main}";s:3:"url";s:36:"/index.php/install/wizard/installDb/";s:11:"script_name";s:10:"/index.php";s:4:"skin";s:7:"default";}
```

Running the install from the shell (php -f install.php -- [...]) fails with another error:

``` php
PHP Fatal error:  Call to a member function getId() on a non-object in /XXX/app/code/core/Mage/Dataflow/Model/Profile.php on line 135
```

**Comment by leesbian** [comment]
unable to replicate the first issue, but yes - the installer will call whatever upgrade routines are necessary to update DB schema with the version of each module included in the distribution

second issue has been replicated - will have a fix shortly

**Comment by leesbian** [comment]
Resolved in commit fb725a3281da17945604d1d4f6097f27f5c1d56f
